packages/web/modals/review-order.tsx

```diff
@@ -98,7 +105,38 @@ export function ReviewOrder({
   const [manualSlippage, setManualSlippage] = useState("");
   const [isEditingSlippage, setIsEditingSlippage] = useState(false);
   const [tab] = useQueryState("tab", parseAsString.withDefault("swap"));
-  const { isOneClickTradingEnabled } = useOneClickTradingSession();
+
+  const [showOneClickTradingSettings, setShowOneClickTradingSettings] =
+    useState(false);
+
+  const {
+    isEnabled: is1CTEnabled,
+    changes: transaction1CTParamsChanges,
+    setChanges: setTransaction1CTParamsChanges,
+    transactionParams: transaction1CTParams,
+    amountSpent: amountSpent1CT,
+    remainingSpendLimit: remaining1CTSpendLimit,
+    setTransactionParams: setTransaction1CTParams,
+    commitSessionChange: commit1CTSessionChange,
+    resetParams: reset1CTParams,
+    commitSessionChangeIsLoading: commit1CTSessionChangeIsLoading,
+  } = useOneClickTradingSessionManager({
+    onCommit: () => {
+      confirmAction();
+      onClose();
+    },
+  });
+
+  const wouldExceedSpendLimit = useMemo(
```

Comment:
thoughts on lifting this calculation into `useOneClickTradingSessionManager`? If you believe it would be useful elsewhere or in the future? Perhaps as a function if outside data is needed

Reply:
hm, good thought

originally didn't want to mix the current swap concerns with session manager, but it also makes sense to encapsulate this logic
let me see if I can find a good place for it

Reply:
found this a nice and cozy new home in 675bb30deac0